includes/admin/class-wc-rest-user-exists-controller.php

```diff
@@ -0,0 +1,77 @@
+<?php
+/**
+ * Class WC_REST_User_Exists_Controller
+ *
+ * @package WooCommerce\Payments\Admin
+ */
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * REST controller to check if a user exists.
+ */
+class WC_REST_User_Exists_Controller extends WP_REST_Controller {
+
+	/**
+	 * Endpoint namespace.
+	 *
+	 * @var string
+	 */
+	protected $namespace = 'wc/v3';
+
+	/**
+	 * Endpoint path.
+	 *
+	 * @var string
+	 */
+	protected $rest_base = 'users/exists';
+
+	/**
+	 * Configure REST API routes.
+	 */
+	public function register_routes() {
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base,
+			[
+				'methods'             => WP_REST_Server::CREATABLE,
+				'callback'            => [ $this, 'user_exists' ],
+				'permission_callback' => '__return_true',
+				'args'                => [
+					'email' => [
+						'required'    => true,
+						'description' => __( 'Email address.', 'woocommerce-payments' ),
+						'type'        => 'string',
+						'format'      => 'email',
+					],
+				],
+			]
+		);
+	}
+
+	/**
+	 * Retrieve if a user exists by email address.
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 *
+	 * @return WP_REST_Response
+	 */
+	public function user_exists( WP_REST_Request $request ): WP_REST_Response {
+		$email   = $request->get_param( 'email' );
+		$user    = get_user_by( 'email', $email );
```

Comment:
This works fine, just an idea/alternative and maybe nit: 
We have this [email_exists()](https://developer.wordpress.org/reference/functions/email_exists/) function which might be lighter to determine whether the user exists than retrieving the whole user object.
This function is also the one used by [wc_create_new_customer()](https://woocommerce.github.io/code-reference/files/woocommerce-includes-wc-user-functions.html#source-view.46) to trigger the error we want to display, so it may be more consistent too.

Reply:
Fixed on https://github.com/Automattic/woocommerce-payments/pull/4436/commits/31a38c144fa7cbd2d447eeb4910c04fe9a4777ee.